yt_dlp/extractor/theguardian.py

```diff
@@ -0,0 +1,134 @@
+            'creator': 'Gaby Hinsliff',
+            'thumbnail': 'md5:28932a7b5a25b057be330d2ed70ea7f3',
+            'release_date': '20231102'
+        }
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        return {
+            'id': video_id,
+            'title': self._og_search_title(webpage) or get_element_by_class('content__headline', webpage),
+            'description': self._og_search_description(webpage),
+            'creator': self._html_search_meta('author', webpage),
+            'thumbnail': self._og_search_thumbnail(webpage),
+            'release_date': unified_strdate(self._html_search_meta('article:published_time', webpage)),
+            'url': extract_attributes(get_element_html_by_class(
+                'podcast__player', webpage) or '').get('data-source'),
+        }
+
+
+class TheGuardianPodcastPlaylistIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?theguardian\.com/\w+/series/(?P<id>[\w-]+)/?(?:$|[?#])'
+    _PAGE_SIZE = 20
+    _TESTS = [{
+        'url': 'https://www.theguardian.com/football/series/theguardianswomensfootballweekly',
+        'info_dict': {
+            'id': 'theguardianswomensfootballweekly',
+            'title': "The Guardian's Women's Football Weekly | Football | The Guardian",
+            'description': 'md5:e2cc021311e582d29935a73614a43f51'
+        },
+        'playlist_mincount': 69
+    }, {
+        'url': 'https://www.theguardian.com/news/series/todayinfocus',
+        'info_dict': {
+            'id': 'todayinfocus',
+            'title': 'Today in Focus | News | The Guardian',
+            'description': 'md5:0f097764fc0d359e0b6eb537be0387e2'
+        },
+        'playlist_mincount': 37
+    }, {
+        'url': 'https://www.theguardian.com/news/series/the-audio-long-read',
+        'info_dict': {
+            'id': 'the-audio-long-read',
+            'title': 'The Audio Long Read | News | The Guardian',
+            'description': 'md5:5462994a27527309562b25b6defc4ef3'
+        },
+        'playlist_mincount': 1008
+    }]
+
+    def _fetch_page(self, url, playlist_id, page):
+        page += 1
+        webpage = self._download_webpage(
+            f'{url}?page={page}', playlist_id, note=f'Downloading page: {page}', expected_status=404)
+
+        episodes = traverse_obj(
+            get_elements_html_by_class('fc-item--type-media', webpage),
+            (..., {extract_attributes}, 'data-id'))
+
+        for entry in episodes:
+            yield self.url_result(urljoin('https://www.theguardian.com', entry), TheGuardianPodcastIE)
+
+    def _real_extract(self, url):
+        podcast_id = self._match_id(url)
+        webpage = self._download_webpage(url, podcast_id)
+
+        title = self._generic_title(url, webpage, default='')
+        description = self._og_search_description(webpage) or get_element_by_class(
+            'header__description', webpage)
```

Comment:
Might be nice to extract a title without the " | The Guardian" junk at the end. Also use `clean_html` just in case
```suggestion
        title = clean_html(get_element_by_class(
            'index-page-header__title', webpage)) or self._generic_title(url, webpage)
        description = self._og_search_description(webpage) or clean_html(get_element_by_class(
            'header__description', webpage))
```

Reply:
If the URL given by the user has a page, the title will be something like: ``Today in Focus | Page 2 of 66 | News | The Guardian``.
Is there a helper function that can be used to handle this? Or is it fine to do something like ``title, _ = title.split('|')` ?

Reply:
`remove_end` is the helper typically used for this, but it needs a fixed string to remove; so it wouldn't be useful here

IMO let's try to grab the clean title from one of these elements instead of doing string surgery with the title element
```py
        title = clean_html(get_element_by_class(
            'index-page-header__title', webpage) or get_element_by_class('flagship-audio__title', webpage))
```

Reply:
I implemented your suggestion bashonly. It seems to pick up all the titles correctly so far.